src/NexusMods.App.UI/Controls/TreeDataGrid/AValueComponent.cs

```diff
@@ -85,10 +85,21 @@ public ValueComponent(
         IObservable<T> valueObservable,
         bool subscribeWhenCreated = false) : base(initialValue, valueObservable, subscribeWhenCreated) { }
 
-    public ValueComponent(
+    public ValueComponent(T value) : base(value) { }
+}
+
+public class ComparableValueComponent<T> : AValueComponent<T>, IItemModelComponent<ComparableValueComponent<T>>, IComparable<ComparableValueComponent<T>>
+where T : IComparable<T>
+{
+    public int CompareTo(ComparableValueComponent<T>? other)
+    {
+        return other is null ? 1 : Value.Value.CompareTo(other.Value.Value);
+    }
+
+    public ComparableValueComponent(
```

Comment:
You don't need to create a new class for this, `ValueComponent` can implement those interfaces. The constructor should have a `IComparer<T>?` argument so that any consumer can specify how comparison of `T` should be done. If you get `null` you can get the default comparer instead in the implementation.

Reply:
Yeah, I wasn't sure whether updating the existing one was best or now. Change